From a59c01682ef5c6c1d2243caea984ef3a077ea472 Mon Sep 17 00:00:00 2001
From: Shunqing Chen <csq@rock-chips.com>
Date: Wed, 16 Dec 2020 06:32:29 +0000
Subject: [PATCH] power: rk817-battery: fix ydsoc calculate error

ydsoc error can lead to charge or discharge become slow.

Signed-off-by: Shunqing Chen <csq@rock-chips.com>
Change-Id: Ie17f0064eda3723378f49caef08bca212c322d70
---
 drivers/power/supply/rk817_battery.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/power/supply/rk817_battery.c b/drivers/power/supply/rk817_battery.c
index 450b80f09532..1680acdaca81 100644
--- a/drivers/power/supply/rk817_battery.c
+++ b/drivers/power/supply/rk817_battery.c
@@ -2424,7 +2424,7 @@ static void rk817_bat_smooth_algorithm(struct rk817_battery_device *battery)
 	}
 
 	/* discharge: sm_linek < 0, if delate_cap <0, ydsoc > 0 */
-	ydsoc = battery->sm_linek * abs(delta_cap / DIV(battery->fcc)) / 10;
+	ydsoc = battery->sm_linek * abs(delta_cap / 10) / DIV(battery->fcc);
 
 	DBG("smooth: ydsoc = %d, fcc = %d\n", ydsoc, battery->fcc);
 	if (ydsoc == 0) {
-- 
2.35.3

